system/jlib/jmetrics.cpp

```diff
@@ -202,6 +194,18 @@ MetricSink *MetricsReporter::getSinkFromLib(const char *type, const char *sinkNa
     return pSink;
 }
 
+// Method for use when testing
+void MetricsReporter::addSink(MetricSink *pSink, const char *name)
+{
+    //
+    // Add the sink if it does not already exist
+    auto sinkIt = sinks.find(name);
+    if (sinkIt == sinks.end())
+    {
+        sinks.insert({std::string(name), std::unique_ptr<SinkInfo>(new SinkInfo(pSink))});
```

Comment:
Will pSink leak?  It looks like that is generally true (for initializeSinks as well).  If this function takes ownership then it should delete pSink if it isn't added.

Reply:
The destructor for the reporter deletes the registered sinks.

Reply:
initializeSinks - agreed - 'sinks' find() match should guarentee the insert succeeds